src/main/java/org/craftercms/studio/impl/v2/repository/blob/s3/StudioAwsS3BlobStore.java

```diff
@@ -361,4 +356,15 @@ public void initialPublish(String siteId) throws SiteNotFoundException {
         }
     }
 
+    @Override
+    public void publishAll(String siteId, String publishingTarget) {
+        Mapping previewMapping = getMapping(publishingTargetResolver.getPublishingTarget());
+        Mapping targetMapping = getMapping(publishingTarget);
+
+        logger.debug("Performing publish all for site {0} to {1}", siteId, targetMapping);
+        deleteFolder(targetMapping.target, targetMapping.prefix, getClient());
```

Comment:
This can be quite expensive. Can we not use AWS Sync mechanics?

Reply:
The Java SDK doesn't provide the `sync` feature from the CLI, the only option is to delete everything and copy it similar to what we do in the git repo